.github/workflows/cd-pipeline.yaml

```diff
@@ -138,24 +152,55 @@ jobs:
           # HACK: Obtain chart tarball out of the latest commit
           chart_version="$(yq ".version" charts/bitnami/${chart}/Chart.yaml)"
           aws s3 cp s3://${{ secrets.AWS_S3_BUCKET }}/bitnami/${chart}-${chart_version}.tgz download/
-          # Rebuild index
-          helm repo index --url https://charts.bitnami.com/bitnami --merge index/bitnami/index.yaml download
-          cp download/index.yaml index/bitnami/index.yaml
+
           cd index
-          # Push changes
           git config user.name "Bitnami Containers"
           git config user.email "bitnami-bot@vmware.com"
-          git add bitnami/index.yaml && git commit -m "${chart}-${chart_version}: Update index.yaml" -s && git push
+
+          attempts=0
+          max_attempts=5
+          is_index_updated=0
+          while [[ $attempts -lt $max_attempts ]]; do
+            attempts=$((attempts + 1))
+            git fetch origin index
+            git reset --hard origin/index
+
+            # Rebuild index
+            helm repo index --url https://charts.bitnami.com/bitnami --merge bitnami/index.yaml ../download
+            cp ../download/index.yaml bitnami/index.yaml
+            # Push changes
+            git add bitnami/index.yaml && git commit -m "${chart}-${chart_version}: Update index.yaml" -s
+            git push && is_index_updated=1 || true
+
+            if [[ $is_index_updated -eq 1 ]]; then
+              echo "::set-output name=result::ok"
+              break
+            else
+              echo "Failed to push during attempt $attempts"
+            fi
+          done
+
+          # If after $max_attempts attempts changes can't be pushed, the action is considered as failed
+          echo "::set-output name=result::fail"
+      - id: show-error
+        name: 'Show error'
+        if: ${{ steps.update-index.outputs.result == 'fail' }}
+        uses: actions/github-script@v3
+        with:
+          script: |
+              core.setFailed('There was a problem updating the index.')
 
   # If the CD Pipeline does not succeed we should notify the interested agents
   slack-notif:
     runs-on: ubuntu-latest
-    needs: vib-publish
+    needs:
+      - vib-publish
+      - update-index
     if: always()
     name: Notify unsuccessful CD run
     steps:
       - name: Notify in Slack channel
-        if: ${{ needs.vib-publish.result != 'success' }}
+        if: ${{ needs.vib-publish.result != 'success' ||  needs.vib-publish.result != 'ok' }}
```

Comment:
Why this difference? `success` vs. `ok`?

Reply:
According to the [documentation](https://docs.github.com/en/actions/learn-github-actions/contexts#needs-context), `ok` is not a valid result

Reply:
My mistake here! This should be `needs.update-index.result`. That way, we will be notified in the Slack channel either if the publishing process **or** the update index one has failed.

Is it clear now? I will update the code accordingly 😁